src/components/ChatEntry.js

```diff
@@ -1,22 +1,46 @@
-import React from 'react';
+import React, { useState } from 'react';
+import TimeStamp from './TimeStamp';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
 
 const ChatEntry = (props) => {
+
+  const [isMessageLiked, setIsMessageLiked] = useState(false)
+
+
+  const toggleLikeButton = (event) => {
+    const updatedChatEntry = {
+      id: props.id,
+      sender: props.sender,
+      body: props.body,
+      timeStamp: props.timeStamp,
+      liked: !props.liked
+    }
+    setIsMessageLiked(!isMessageLiked);
+    props.onHandleLikes(updatedChatEntry);
+    };
```

Comment:
You're already keeping track of chatEntryData in app.js (updateChatEntryData on line 24) so we should keep all chatEntryData logic in app.js instead of having it in different components which makes the code hard to maintain. It's easier to remember that everything is in app.js (instead of having to check each component to see if make changes to chatEntryData). In industry, an application could have hundreds of components so keeping logic organized is important. 

We also discussed lifting up state in the Learn readings. Here, we can lift up the like state out of ChatEntry and into App.js to follow this pattern of lifting state up.

This means toggleLikeButton should live in app.js and get passed down to ChatEntry as a prop. So you'd have a method like toggleLike in app.js:
```javascript
const toggleLike = (id) => {
    setChatData((chatData) =>
    chatData.map((message) => {
      if (message.id === id) {
        return {...message, liked: !message.liked};
      } else {
        return message;
      }
    })
    );
  };
```

and then your onClick event handler on line 29 below would look like: 
```javascript
onClick={() => props.toggleLike(props.id)}
```